packages/starknet-snap/src/utils/formatterUtils.ts

```diff
@@ -7,3 +9,33 @@ export const hexToString = (hexStr) => {
   }
   return str;
 };
+
+/**
+ * Maps deprecated parameters to their new equivalents in the requestParams object
+ * and removes the deprecated parameters afterward.
+ *
+ * @param requestParams - The object containing the API request parameters.
+ * @param mappings - A record of key-value pairs where the key is the old parameter
+ * and the value is the new parameter.
+ * @example
+ * const paramMappings = {
+ *   signerAddress: 'address',
+ *   senderAddress: 'address',
+ *   txnInvocation: 'calls',
+ *   invocationsDetails: 'details',
+ *   transaction: 'details'
+ * };
+ * mapDeprecatedParams(apiParams.requestParams, paramMappings);
+ */
+export const mapDeprecatedParams = (
+  requestParams: Record<string, Json>,
+  mappings: Record<string, string>,
+) => {
+  Object.keys(mappings).forEach((oldParam) => {
+    const newParam = mappings[oldParam];
+    if (Object.prototype.hasOwnProperty.call(requestParams, oldParam)) {
+      requestParams[newParam] = requestParams[oldParam];
+      delete requestParams[oldParam]; // Remove old param after mapping
+    }
+  });
+};
```

Comment:
```suggestion
export const mapDeprecatedParams = <Params>(
  requestParams: Params
  mappings: Record<string, string>,
) => {
  Object.keys(mappings).forEach((oldParam) => {
    const newParam = mappings[oldParam]  as unknown as keyof Params;
    if (Object.prototype.hasOwnProperty.call(requestParams, oldParam)) {
      requestParams[newParam] = requestParams[oldParam];
      delete requestParams[oldParam]; // Remove old param after mapping
    }
  });
};
```

nit:
in that case, we dont need to cast the type in execute txn